### ISSUE TITLE
Proform 5000 treadmill connection issue

### ISSUE THREAD

**Issue by LJP74** [cagnulein/qdomyos-zwift#1448]
Hi guys
I have a Proform 5000 treadmill and I'm just having trouble getting the treadmill info to reach the QZ app. (app is running on an iPad (16.4.1)

I have followed the install guide and have QZ Companion installed on the treadmill (android tablet in privilege mode)
QZ companion setting is 'other'

I restart the treadmill and hop into iFIt and start a manual session.

In my QZ app, I've entered the treadmill IP and I'm seeing the tiles but no data is reaching the app from the treadmill.
My current setting is 'Nordictrack 10' enabled and the treadmill IP is in the 'Nordictrack 2950 IP' field.
I did have one instance under this setting where it did connect and the data was reaching the tiles but since then no luck.
I have tried all other settings in the 'Proform/Nordictrack Options' menu but have had no luck.

Not sure if the Proform 5000 is the issue as I've not come across anybody using this particular model with QZ.

I've attached a debug file.

Thanks in advance and really appreciate the support you give the app. Awesome work!

Thanks
[debug-Mon_May_1_12_59_54_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11363267/debug-Mon_May_1_12_59_54_2023.log)

**Comment by cagnulein** [comment]
Hi  @LJP74 the good news is that I'm seeing in the log the communication between the companion and qz, good!
So now we have to understand why the companion is sending always 0.
Are you able to attach here a debug log from the companion?
You can grab it clicking on the dump log button on the companion after a short manual workout and then you will see the logcat.txt file in the /sdcard/ of  your tablet
Let me know

**Comment by LJP74** [comment]
Thanks so much for the quick reply.
I've attached two screengrabs of the tablet -  am I in the right area?
![56F6BB83-75E5-49A8-B7D1-33C5A8EC2485_1_105_c](https://user-images.githubusercontent.com/90441632/235404480-828be4f5-5407-4e70-9f4e-0ad7e8ebb642.jpeg)
![F05F71AA-06D0-437B-A778-23A0BD1694F6_1_105_c](https://user-images.githubusercontent.com/90441632/235404486-51c01dd7-ec5b-466d-82d7-2cbbdf35153c.jpeg)

**Comment by cagnulein** [comment]
mmm nope but send me these files also, they are useful too

**Comment by LJP74** [comment]
Hi again, I think I've hit a roadblock. For the life of me, I can't seem to get a logcat.txt file to show up on the treadmills sd card. Have tried pretty much everything I can think of but I'll keep trying.

**Comment by cagnulein** [comment]
Ok send me the files in your above screenshot in the meantime. They could be useful too

**Comment by LJP74** [comment]
Managed to get a log file created - is this any good?
[logcat.log](https://github.com/cagnulein/qdomyos-zwift/files/11363841/logcat.log)

**Comment by cagnulein** [comment]
it doesn't have any qz companion log here, very strange. are you sure the companion is running? as I said send me also the log in the above screenshot in the wolflog directory, maybe they have a different format

**Comment by cagnulein** [comment]
also @LJP74 check if the qz companion has all the permission granted in the android settings of your tablet

**Comment by LJP74** [comment]
Thank you for your patience on this :)
I rebooted the treadmill and it fires up with QZ companion first and then iFit (with QZ running in the background)
I did a manual workout in iFit and then a dump log in QZ Companion app on the treadmill tablet.
Attached this log below but not sure this is showing any activity from QZ at all?

Also checked the treadmill tablet QZ companion app system privileges and attached a couple of screenshots of what is showing. again, I'm not sure if this looks correct?
Have also attached log form the wold folder on the sd card.
Thanks again - really appreciate your help.

![IMG_1786](https://user-images.githubusercontent.com/90441632/235418195-058ab387-cc2d-4b4c-bcde-b6ff06971b15.jpeg)
![IMG_1787](https://user-images.githubusercontent.com/90441632/235418217-09e22d41-dbab-4a34-95a7-d5fb7ccd462e.jpeg)
[logcat.log](https://github.com/cagnulein/qdomyos-zwift/files/11363944/logcat.log)
[2023-05-01_logs.txt](https://github.com/cagnulein/qdomyos-zwift/files/11363945/2023-05-01_logs.txt)

**Comment by cagnulein** [comment]
ok permissions look fine.
the https://github.com/cagnulein/qdomyos-zwift/files/11363945/2023-05-01_logs.txt shows no metrics on it. i guess that's the main issue.
did you stop the workout before collecting this log?
@victorypoint said to me that stopping the workout could corrupt the log file, so try also without stopping the workout (collecting it while the workout is running).

I guess we are near

**Comment by LJP74** [comment]
Hi there
Have created the log without stopping the manual workout. Attached file :)
[logcat.log](https://github.com/cagnulein/qdomyos-zwift/files/11364219/logcat.log)

**Comment by cagnulein** [comment]
ok there is for sure an issue. Your ifit app doesn't write the metric in the file. That's the main issue.
Maybe you have other files in the android tablet? We are looking files that contains something like "changed speed to"...

@victorypoint any idea?

**Comment by LJP74** [comment]
Ok, I’ll have a look around for the other files or maybe a factory reset and another install of QZ companion might work. Thanks again

**Comment by cagnulein** [comment]
ok let me know! also share the current ifit app version, it could be useful to @victorypoint

**Comment by victorypoint** [comment]
@LJP74 , I think its just a matter of finding the correct files where your treadmill metrics are saved. Typically they are in a hidden folder thats not visible in most file managers - the hidden folder would start with a . like /sdcard/.wolflogs. Are you able to execute shell commands on the treadmill via ADB? If so, run this:

adb shell ls -R -al /sdcard/ > resursive.txt

And upload the resulting recursive text file so we can inspect.

**Comment by LJP74** [comment]
Great - thanks guys. Will have a look today and will let you know how I go.

**Comment by LJP74** [comment]
Success!! I did a factory reset of the treadmill and re-installed QZ Companion and it's now working.

The major difference on this install was that I managed to get an ADB connection to the treadmill when installing QZ Companion. (from a windows laptop)
On the previous install, I used the apk method from the treamills android tablet.
Maybe this was the reason for the metrics not beings sent etc.

Anyway, super pleased with the set-up and have also got Zwift running.(linked to the QZ Wahoo connection)

Thanks to @cagnulein and @victorypoint for all your help and patience - really appreciate your time and support.
Thanks guys
